Magento_Theme/web/js/custom-modal.js

```diff
@@ -0,0 +1,44 @@
+define(['jquery', 'jquery-ui-modules/widget', 'mage/cookies'], function ($) {
+  $.widget('vaimo.modalWidget', {
+    options: {
+      modalOpenClass: 'modal',
+      modalNotOpenClass: 'not-open-modal',
+      modalShadowClass: 'modal-shadow',
+    },
+
+    _create: function () {
+      if (!$.mage.cookies.get('modalClosed')) {
+        this.element.css('display', 'block');
+        this.addModalShadow();
+      }
+      this._on({
+        'click #close-modal-icon': this.closeModal,
+        'click .modal__close-btn': this.closeModal,
+        'submit .modal__form': this.submitForm,
+      });
+    },
+
+    submitForm: function () {
+      this.sendModalCookie();
+    },
+
+    sendModalCookie: function () {
+      if (!$.mage.cookies.get('modalClosed')) {
```

Comment:
pro tip: use revert statements, not to make next level nesting of braces. it could be difficult first time but then it has to do your code such better
```js
if ($.mage.cookies.get('modalClosed')) {
  return this;
}

$.mage.cookies.set('modalClosed', true, {
  lifetime: 60 * 60 * 24 * 7,
});
```